Tencent.WeType 1.2.2.630 and change some properties

stevenlele opened the pull request:
Upgrade behavior is changed to "deny" because the silent installation won't terminate the old version, causing the update to not take effect until the user manually reboots the computer. Instead, users should use the software's built-in updater, or run the installer interactively. "Uninstall previous" is not ideal because it clears user data.

The built-in updater uses the "/Silence" switch (and internally it terminates the old version), which has the same effect as "/s". Other switches like "/S" (default of nullsoft) or "/random" will result in an incomplete installation somehow.

---

Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/165220)

stevenlele committed:
Tencent.WeType 1.2.2.630

hackean-msft commented:
/azp run

hackean-msft commented:
/azp run

stephengillie commented:
Automatic Validation ended with:
> No errors to post.

(Automated response - build 887.)

stephengillie commented on manifests/t/Tencent/WeType/1.2.2.630/Tencent.WeType.installer.yaml:
```diff
@@ -0,0 +1,26 @@
+  DisplayVersion: 1.2.2.630
+ElevationRequirement: elevatesSelf
+Installers:
+- Architecture: x86
```
```suggestion
- Architecture: x64
```

stephengillie reviewed (changes requested):
Hi @stevenlele,

From the previous error, I'm guessing this might be an `x64` package. Our `x86` validations are performed on 32-bit OSes.

stevenlele committed:
Correct installer architecture

stephengillie commented:
> @stephengillie Could you check what's the SHA256 of the file that the pipeline downloaded?

Sorry, but I don't have any more access to the pipeline VMs than you do. I don't know if the file hash is emitted in the artifact logs, do you?

stephengillie commented:
Automatic Validation ended with:

> Installation failed with exit code -1978335215
> 2024-07-31 18:48:57.752 [CLI ] Package hash verification failed. SHA256 in manifest [af04e13b073a993c1e16f48b6fc06db7d24875bbe66e182a8dd64e62ab571790] does not match download [17e98d047d5b4ed5a84472b7f7b9988e6511264673ee58220edfa80c15ba4df3]

Automated error analysis suggests -1978335215 may mean The installer files hash does not match the manifest

(Automated response - build 887.)

stephengillie commented:
/azp run

stevenlele committed:
Update properties

Summary:
- InstallerType=exe: Installer is heavily modified Nullsoft (NSIS), only `/s` and `/Silence` are valid, `/S` is broken
- UpgradeBehavior=deny: Silent update is broken and requires reboot, either use in-app updater or run installer interactively
- Removed AppsAndFeaturesEntries: It matches the package name thus unnecessary; `wingetcreate update` won't update the `DisplayVersion` correctly
- Removed Moniker: it's the same as package name

stephengillie commented:
Hi @stevenlele,

This manifest removes Apps and Features entries that are present in previous PR versions. These entries should be added to this version, to maintain version matching, and prevent the 'upgrade always available' situation with this package.

(Automated response - build 887.)
<!--
[Policy] Needs-Author-Feedback
-->

stephengillie commented:
Hi @stevenlele,

This manifest removes Apps and Features entries that are present in previous PR versions. These entries should be added to this version, to maintain version matching, and prevent the 'upgrade always available' situation with this package.

(Automated response - build 887.)
<!--
[Policy] Needs-Author-Feedback
-->

stephengillie commented:
Thanks for the thorough background. 
- If the `DisplayVersion` matches the `PackageVersion` in every manifest, then the `DisplayVersion` should be removed. Part of this is to reduce redundant data in the manifest, and another is because the `DisplayVersion` can cause issues both with uploading other manifest versions, and also the "upgrade always available" situation. 
- The `DisplayName` is much less impactful and should probably remain with those package versions. 
- I'm unsure with the `Publisher` field, as I wasn't aware of it as an `AppsAndFeaturesEntries` parameter, but it's probably as innocuous as the `DisplayName`.

 Since the `DisplayVersion` apparently matches the `PackageVersion` in previous manifests, I believe that this is okay to merge without these parameters. But if the "upgrade always available" situation occurs, then the `DisplayVersion` is likely the issue, and would need to be either added to this manifest, or removed from previous manifests.

stephengillie commented:
Hi @stevenlele,

This manifest removes Apps and Features entries that are present in previous PR versions. These entries should be added to this version, to maintain version matching, and prevent the 'upgrade always available' situation with this package.

(Automated response - build 887.)
<!--
[Policy] Needs-Author-Feedback
-->
